pykubectl/kubectl.py

```diff
@@ -3,26 +3,27 @@
 from subprocess import CalledProcessError, check_output
 
 
-class KubeCtl(object):
+class KubeCtl:
     def __init__(self, bin='kubectl', global_flags=''):
-        super(KubeCtl, self).__init__()
-        self.kubectl = '{} {}'.format(bin, global_flags)
+        super().__init__()
+        self.kubectl = f'{bin} {global_flags}'
 
     def execute(self, command, definition=None, safe=False):
-        cmd = '{} {}'.format(self.kubectl, command)
+        cmd = f'{self.kubectl} {command}'
 
         if definition:
-            pre = 'echo \'{}\''.format(definition)
-            cmd = '{} | {} -f -'.format(pre, cmd)
+            cmd = f'cat <<EOF | {cmd} -f -\n' \
+                  f'{definition}\n' \
+                  f'EOF'
```

Comment:
this might not be resilient if the definition contains EOF. I suggest you shlex, or you save to a temp file using the [tempfile](https://docs.python.org/3/library/tempfile.html) module

Reply:
changed to use tempfile in latest